climada_petals/hazard/copernicus_interface/time_utils.py

```diff
@@ -0,0 +1,156 @@
+        if 1 <= month <= 12:
+            return month
+        else:
+            raise ValueError("Month number must be between 1 and 12.")
+    if isinstance(month, str):
+        if not month.strip():
+            raise ValueError("Month cannot be empty.")  # e.g. "" or "   "
+        month = month.capitalize()  # Ensure consistent capitalization
+        if month in calendar.month_name:
+            return list(calendar.month_name).index(month)
+        elif month in calendar.month_abbr:
+            return list(calendar.month_abbr).index(month)
+    raise ValueError(f"Invalid month input: {month}")
+
+
+def calculate_leadtimes(year, initiation_month, valid_period):
+    """
+    Calculate lead times in hours for a forecast period based on initiation and valid months.
+
+    This function computes a list of lead times (in hours) for a seasonal forecast, starting
+    from the initiation month to the end of the valid period. The lead times are generated
+    in 6-hour steps, following the standard forecast output intervals.
+
+    Parameters
+    ----------
+    year : int
+        Year of the forecast initiation.
+    initiation_month : int or str
+        Initiation month of the forecast, as integer (1-12) or month name (e.g., 'March').
+    valid_period : list of int or str
+        List containing the start and end month of the valid period, either as integers (1-12)
+        or month names (e.g., ['June', 'August']). Must contain exactly two elements.
+
+    Returns
+    -------
+    list of int
+        List of lead times in hours, sorted and spaced by 6 hours.
+
+    Raises
+    ------
+    ValueError
+        If initiation month or valid period months are invalid or reversed.
+
+    Notes
+    -----
+    - The valid period may extend into the following year if the valid months are after December.
+    - Lead times are calculated relative to the initiation date.
+    - Each lead time corresponds to a 6-hour forecast step.
+
+    Example:
+    ---------
+    If the forecast is initiated in **December 2022** and the valid period is **January
+    to February 2023**,
+    the function will:
+    - Recognize that the forecast extends into the next year (2023).
+    - Compute lead times starting from **December 1, 2022** (0 hours) to **February 28, 2023**.
+    - Generate lead times in 6-hour intervals, covering the entire forecast period from
+    December 2022 through February 2023.
+    """
+
+    # Convert initiation month to numeric if it is a string
+    if isinstance(initiation_month, str):
+        initiation_month = month_name_to_number(initiation_month)
+
+    # Convert valid_period to numeric
+    valid_period = [
+        month_name_to_number(month) if isinstance(month, str) else month
+        for month in valid_period
+    ]
+
+    # We expect valid_period = [start, end]
+    start_month, end_month = valid_period
+
+    # Immediately check for reversed period
+    if end_month < start_month:
```

Comment:
What if `start_month` is December (2024) and `end_month` is January (2025) ? the function will raise an error and not even execute the following code which is supposed to check for this.